src/Microsoft.TemplateEngine.Core/Operations/PhasedOperation.cs

```diff
@@ -37,6 +35,12 @@ public IOperation GetOperation(Encoding encoding, IProcessorState processorState
                 while (currentSource?.MoveNext() ?? false)
                 {
                     Phase c = currentSource.Current;
+
```

Comment:
There's nothing analogous to the `c == null` check in the existing code. It seems like it can't happen unless there is a bug in the enumerator - MoveNext() should return false in this case.